envoy/upstream/upstream.h

```diff
@@ -1131,7 +1131,8 @@ class ClusterInfo : public Http::FilterChainFactory {
   /**
    * @return std::shared_ptr<UpstreamLocalAddressSelector> as upstream local address selector.
    */
-  virtual UpstreamLocalAddressSelectorConstSharedPtr getUpstreamLocalAddressSelector() const PURE;
+  virtual const UpstreamLocalAddressSelectorConstSharedPtr
```

Comment:
Returning a const value doesn't make any sense I don't think; it doesn't have any effect.

Reply:
You are correct, I've mistakenly changed it when I saw the const method. It is already returning a shared-ptr to a const value, so it should be sufficient.